docs/end-user/stack-samples-and-call-trees.md

```diff
@@ -0,0 +1,55 @@
+
+This image shows that the profiler sampled `B` at the beginning and end of the profiling. This will not change the shape of the call tree, but it will change the self and running times reported.
+
+![This graphic demonstrates the modified call tree. It is a chart of the stacks A, B, C, and doWork, each connected by an arrow going from the root A to the leaf doWork. A has a running time of 5ms, and a self time of 0ms. B has a running time of 5ms, and a self time of 2ms. C has a running time of 3ms, and a self time of 0ms. doWork has a running time of 3ms, and a self time of 3ms.](../images/simple-call-tree-self-time.svg)
+
+The call node with the function `B` continues to have the running time of 5ms, but also has a self time of 2ms. Function `C` has a reduced running time of `3ms` and finally do work was only observed `3ms` for both the running and self time.
+
+In summary, self time is where work was actually happening when observed by the profiler.
+
+## How to form a call tree
+
+> Note: Feel free to follow along with a real call tree that reproduces this structure: [https://perfht.ml/2w45IdC](https://perfht.ml/2w45IdC)
+
+The simple call tree above did not take into account any branching stacks. It only concerned itself with running and self time. This section will dive into how the call tree handles stacks of different shapes.
+
+![This image shows contiguous stacks, and how they are sampled over time. It fades out as it goes left to right to indicate that samples are gathered for long periods of time. The stacks are labeled by 1 millisecond intervals. The stacks are labeled by functions that are single capital letters, ranged A to H.](../images/samples.svg)
+
+Consider the first 3 samples in this profile.
+
+![On the left of this graph are three samples with different stacks. A, B, C, D, E. A, B, C, F, G. A, B, H, F, G. The bottom row with A is all grouped together. The second row with B is all grouped together. The third row has the two C functions grouped together, and the H distinct. All the rest of the function sare not grouped. The right graphic demonstrates the shape of the call tree.](../images/call-tree.svg)
+
+These three samples share things in common. The root of each stack is the function `A`. These could be aggregated together into a single node. The same happens for node `B`. However the next level with the function `C` and `H` differ. At this point the call tree splits into two different nodes. There is a node for the function `C`, and the function `H`.
+
+The call tree will continue merging functions that are at the same level, but only if they are part of the current branching pattern. For instance, consider function `F` and `G`. While both functions are at the same level of the stack, they are at different branching points. The middle `F -> G` stacks have `C` as their ancestor, and the right `F -> G` stacks have `H` as their ancestor, and thus are not grouped together.
```

Comment:
I miss the explanation where the sibling nodes at one level are ordered according to their running time (so, the time including their own children)

(can be added in a subsequent PR)